dashboard/lib/widgets/task_grid.dart

```diff
@@ -71,7 +71,7 @@ class TaskGridContainer extends StatelessWidget {
 /// Results are displayed in a matrix format. Rows are commits and columns
 /// are the results from tasks.
 class TaskGrid extends StatefulWidget {
-  const TaskGrid({
+  TaskGrid({
```

Comment:
Can we avoid removing the const? This is one of the more expensive operations to do in the app, so removing it will slow the performance down.

Could we instead make _processCommitStatuses available with @visbileForTesting and verify the LatticeCells instead?

Reply:
This was a tricky one to navigate since making it const means I cannot have fields that be created after instantiation. It requires everything be final and since we get the sorted list after that doesn't really work in this case. I also remember the lattice cells not having the data to verify but I'll give a second look.

Reply:
If it's not a can of spaghetti, I would try adding the task name to the LatticeCell as that'll make future tests easier to write as well. Otherwise, we could refactor _processCommitStatuses to have another method it calls like `sortCommitStatuses(List<CommitStatus)` that the test can use

Reply:
@visibleForTesting does not seem to expose private members. So not sure what the point of this annotation is.

Reply:
You need to make the member public as well. Then the linter will ensure other classes don't rely on it, but tests can use it.

```dart
@visibleForTesting
List<LatticeCell> processCommitStatuses(...);
```

Reply:
A better solution you mentioned was adding the taskName to the LatticeCell. This was easier and did not result in me having to remove const and working around dart's enforcement of fields with const objects.